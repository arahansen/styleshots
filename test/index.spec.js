const styleshots = require('../index')
const path = require('path')

const url = `file://${path.resolve(__dirname, './fixtures/basic.html')}`

styleshots({
  url,
  selectors: [
    '.plain-div',
    '.red',
    '.always-green .red',
    '.padding',
    '.margin',
    '.border',
    '.rgba',
    '.inline-color',
    {
      selector: '.hover-el',
      hover: true,
    },
    {
      selector: '.active-el',
      active: true,
    },
    {
      selector: '.focus-el',
      focus: true,
    },
  ],
})
