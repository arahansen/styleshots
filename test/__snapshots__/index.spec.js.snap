// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic ".active-el" 1`] = `
Object {
  "blockSize": "25px",
  "cssFloat": "none",
  "display": "block",
  "height": "25px",
  "inlineSize": "25px",
  "length": 283,
  "perspectiveOrigin": "12.5px 12.5px",
  "transformOrigin": "12.5px 12.5px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "25px",
  "webkitLogicalWidth": "25px",
  "width": "25px",
}
`;

exports[`Basic ".active-el:active" 1`] = `
Object {
  "blockSize": "25px",
  "borderBottomStyle": "ridge",
  "borderBottomWidth": "10px",
  "borderLeftStyle": "ridge",
  "borderLeftWidth": "10px",
  "borderRightStyle": "ridge",
  "borderRightWidth": "10px",
  "borderStyle": "ridge",
  "borderTopStyle": "ridge",
  "borderTopWidth": "10px",
  "borderWidth": "10px",
  "cssFloat": "none",
  "display": "block",
  "height": "25px",
  "inlineSize": "25px",
  "length": 283,
  "perspectiveOrigin": "22.5px 22.5px",
  "transformOrigin": "22.5px 22.5px",
  "webkitBorderAfter": "10px ridge rgb(0, 0, 0)",
  "webkitBorderAfterStyle": "ridge",
  "webkitBorderAfterWidth": "10px",
  "webkitBorderBefore": "10px ridge rgb(0, 0, 0)",
  "webkitBorderBeforeStyle": "ridge",
  "webkitBorderBeforeWidth": "10px",
  "webkitBorderEnd": "10px ridge rgb(0, 0, 0)",
  "webkitBorderEndStyle": "ridge",
  "webkitBorderEndWidth": "10px",
  "webkitBorderStart": "10px ridge rgb(0, 0, 0)",
  "webkitBorderStartStyle": "ridge",
  "webkitBorderStartWidth": "10px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "25px",
  "webkitLogicalWidth": "25px",
  "width": "25px",
}
`;

exports[`Basic ".always-green .red" 1`] = `
Object {
  "backgroundColor": "rgb(0, 128, 0)",
  "blockSize": "100px",
  "cssFloat": "none",
  "display": "block",
  "height": "100px",
  "inlineSize": "100px",
  "length": 283,
  "perspectiveOrigin": "50px 50px",
  "transformOrigin": "50px 50px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "100px",
  "webkitLogicalWidth": "100px",
  "width": "100px",
}
`;

exports[`Basic ".border" 1`] = `
Object {
  "blockSize": "0px",
  "borderBottomColor": "rgb(255, 255, 0)",
  "borderBottomStyle": "solid",
  "borderBottomWidth": "3px",
  "borderLeftColor": "rgb(255, 192, 203)",
  "borderLeftStyle": "dashed",
  "borderLeftWidth": "4px",
  "borderRightColor": "rgb(128, 0, 128)",
  "borderRightStyle": "dashed",
  "borderRightWidth": "2px",
  "borderStyle": "solid dashed",
  "borderTopColor": "rgb(0, 0, 255)",
  "borderTopStyle": "solid",
  "borderTopWidth": "1px",
  "borderWidth": "1px 2px 3px 4px",
  "cssFloat": "none",
  "display": "block",
  "height": "0px",
  "inlineSize": "1258px",
  "length": 283,
  "perspectiveOrigin": "632px 2px",
  "transformOrigin": "632px 2px",
  "webkitBorderAfter": "3px solid rgb(255, 255, 0)",
  "webkitBorderAfterColor": "rgb(255, 255, 0)",
  "webkitBorderAfterStyle": "solid",
  "webkitBorderAfterWidth": "3px",
  "webkitBorderBefore": "1px solid rgb(0, 0, 255)",
  "webkitBorderBeforeColor": "rgb(0, 0, 255)",
  "webkitBorderBeforeStyle": "solid",
  "webkitBorderBeforeWidth": "1px",
  "webkitBorderEnd": "2px dashed rgb(128, 0, 128)",
  "webkitBorderEndColor": "rgb(128, 0, 128)",
  "webkitBorderEndStyle": "dashed",
  "webkitBorderEndWidth": "2px",
  "webkitBorderStart": "4px dashed rgb(255, 192, 203)",
  "webkitBorderStartColor": "rgb(255, 192, 203)",
  "webkitBorderStartStyle": "dashed",
  "webkitBorderStartWidth": "4px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "0px",
  "webkitLogicalWidth": "1258px",
  "width": "1258px",
}
`;

exports[`Basic ".container-with-nesting *": parent > child > grandchild 1`] = `
Object {
  "cssFloat": "none",
  "display": "none",
  "fontFamily": "sans-serif",
  "length": 283,
  "perspectiveOrigin": "50% 50%",
  "transformOrigin": "50% 50%",
  "webkitLocale": "\\"en\\"",
}
`;

exports[`Basic ".container-with-nesting *": parent > child 1`] = `
Object {
  "blockSize": "0px",
  "borderBottomLeftRadius": "7px",
  "borderBottomRightRadius": "7px",
  "borderTopLeftRadius": "7px",
  "borderTopRightRadius": "7px",
  "cssFloat": "none",
  "display": "block",
  "fontFamily": "sans-serif",
  "height": "0px",
  "inlineSize": "1264px",
  "length": 283,
  "perspectiveOrigin": "632px 0px",
  "transformOrigin": "632px 0px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "0px",
  "webkitLogicalWidth": "1264px",
  "width": "1264px",
}
`;

exports[`Basic ".container-with-nesting *": parent 1`] = `
Object {
  "blockSize": "0px",
  "cssFloat": "none",
  "display": "block",
  "fontFamily": "sans-serif",
  "height": "0px",
  "inlineSize": "1264px",
  "length": 283,
  "perspectiveOrigin": "632px 0px",
  "transformOrigin": "632px 0px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "0px",
  "webkitLogicalWidth": "1264px",
  "width": "1264px",
}
`;

exports[`Basic ".focus-el" 1`] = `
Object {
  "blockSize": "25px",
  "borderBottomColor": "rgb(144, 77, 33)",
  "borderLeftColor": "rgb(144, 77, 33)",
  "borderRightColor": "rgb(144, 77, 33)",
  "borderTopColor": "rgb(144, 77, 33)",
  "caretColor": "rgb(144, 77, 33)",
  "color": "rgb(144, 77, 33)",
  "columnRuleColor": "rgb(144, 77, 33)",
  "cssFloat": "none",
  "cursor": "pointer",
  "height": "25px",
  "inlineSize": "25px",
  "length": 283,
  "outlineColor": "rgb(144, 77, 33)",
  "textDecorationColor": "rgb(144, 77, 33)",
  "textDecorationLine": "underline",
  "webkitBorderAfter": "0px none rgb(144, 77, 33)",
  "webkitBorderAfterColor": "rgb(144, 77, 33)",
  "webkitBorderBefore": "0px none rgb(144, 77, 33)",
  "webkitBorderBeforeColor": "rgb(144, 77, 33)",
  "webkitBorderEnd": "0px none rgb(144, 77, 33)",
  "webkitBorderEndColor": "rgb(144, 77, 33)",
  "webkitBorderStart": "0px none rgb(144, 77, 33)",
  "webkitBorderStartColor": "rgb(144, 77, 33)",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "25px",
  "webkitLogicalWidth": "25px",
  "webkitTextDecorationsInEffect": "underline",
  "webkitTextEmphasisColor": "rgb(144, 77, 33)",
  "webkitTextFillColor": "rgb(144, 77, 33)",
  "webkitTextStrokeColor": "rgb(144, 77, 33)",
  "width": "25px",
}
`;

exports[`Basic ".focus-el:focus" 1`] = `
Object {
  "blockSize": "25px",
  "borderBottomColor": "rgb(150, 120, 150)",
  "borderLeftColor": "rgb(150, 120, 150)",
  "borderRightColor": "rgb(150, 120, 150)",
  "borderTopColor": "rgb(150, 120, 150)",
  "caretColor": "rgb(150, 120, 150)",
  "color": "rgb(150, 120, 150)",
  "columnRuleColor": "rgb(150, 120, 150)",
  "cssFloat": "none",
  "cursor": "pointer",
  "height": "25px",
  "inlineSize": "25px",
  "length": 283,
  "outlineColor": "rgb(59, 153, 252)",
  "outlineStyle": "auto",
  "outlineWidth": "5px",
  "textDecorationColor": "rgb(150, 120, 150)",
  "textDecorationLine": "underline",
  "webkitBorderAfter": "0px none rgb(150, 120, 150)",
  "webkitBorderAfterColor": "rgb(150, 120, 150)",
  "webkitBorderBefore": "0px none rgb(150, 120, 150)",
  "webkitBorderBeforeColor": "rgb(150, 120, 150)",
  "webkitBorderEnd": "0px none rgb(150, 120, 150)",
  "webkitBorderEndColor": "rgb(150, 120, 150)",
  "webkitBorderStart": "0px none rgb(150, 120, 150)",
  "webkitBorderStartColor": "rgb(150, 120, 150)",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "25px",
  "webkitLogicalWidth": "25px",
  "webkitTextDecorationsInEffect": "underline",
  "webkitTextEmphasisColor": "rgb(150, 120, 150)",
  "webkitTextFillColor": "rgb(150, 120, 150)",
  "webkitTextStrokeColor": "rgb(150, 120, 150)",
  "width": "25px",
}
`;

exports[`Basic ".hover-el" 1`] = `
Object {
  "backgroundColor": "rgb(0, 50, 200)",
  "blockSize": "250px",
  "cssFloat": "none",
  "display": "block",
  "height": "250px",
  "inlineSize": "250px",
  "length": 283,
  "perspectiveOrigin": "125px 125px",
  "transformOrigin": "125px 125px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "250px",
  "webkitLogicalWidth": "250px",
  "width": "250px",
}
`;

exports[`Basic ".hover-el:hover" 1`] = `
Object {
  "backgroundColor": "rgb(0, 155, 200)",
  "blockSize": "250px",
  "cssFloat": "none",
  "display": "block",
  "height": "250px",
  "inlineSize": "250px",
  "length": 283,
  "perspectiveOrigin": "125px 125px",
  "transformOrigin": "125px 125px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "250px",
  "webkitLogicalWidth": "250px",
  "width": "250px",
}
`;

exports[`Basic ".inline-color" 1`] = `
Object {
  "blockSize": "18px",
  "borderBottomColor": "rgb(128, 128, 128)",
  "borderLeftColor": "rgb(128, 128, 128)",
  "borderRightColor": "rgb(128, 128, 128)",
  "borderTopColor": "rgb(128, 128, 128)",
  "caretColor": "rgb(128, 128, 128)",
  "color": "rgb(128, 128, 128)",
  "columnRuleColor": "rgb(128, 128, 128)",
  "cssFloat": "none",
  "display": "block",
  "height": "18px",
  "inlineSize": "1264px",
  "length": 283,
  "outlineColor": "rgb(128, 128, 128)",
  "perspectiveOrigin": "632px 9px",
  "textDecorationColor": "rgb(128, 128, 128)",
  "transformOrigin": "632px 9px",
  "webkitBorderAfter": "0px none rgb(128, 128, 128)",
  "webkitBorderAfterColor": "rgb(128, 128, 128)",
  "webkitBorderBefore": "0px none rgb(128, 128, 128)",
  "webkitBorderBeforeColor": "rgb(128, 128, 128)",
  "webkitBorderEnd": "0px none rgb(128, 128, 128)",
  "webkitBorderEndColor": "rgb(128, 128, 128)",
  "webkitBorderStart": "0px none rgb(128, 128, 128)",
  "webkitBorderStartColor": "rgb(128, 128, 128)",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "18px",
  "webkitLogicalWidth": "1264px",
  "webkitTextEmphasisColor": "rgb(128, 128, 128)",
  "webkitTextFillColor": "rgb(128, 128, 128)",
  "webkitTextStrokeColor": "rgb(128, 128, 128)",
  "width": "1264px",
}
`;

exports[`Basic ".margin" 1`] = `
Object {
  "blockSize": "0px",
  "cssFloat": "none",
  "display": "block",
  "height": "0px",
  "inlineSize": "1250px",
  "length": 283,
  "marginBottom": "7px",
  "marginLeft": "8px",
  "marginRight": "6px",
  "marginTop": "5px",
  "perspectiveOrigin": "625px 0px",
  "transformOrigin": "625px 0px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "0px",
  "webkitLogicalWidth": "1250px",
  "webkitMarginAfter": "7px",
  "webkitMarginBefore": "5px",
  "webkitMarginEnd": "6px",
  "webkitMarginStart": "8px",
  "width": "1250px",
}
`;

exports[`Basic ".padding" 1`] = `
Object {
  "blockSize": "0px",
  "cssFloat": "none",
  "display": "block",
  "height": "0px",
  "inlineSize": "1258px",
  "length": 283,
  "paddingBottom": "3px",
  "paddingLeft": "4px",
  "paddingRight": "2px",
  "paddingTop": "1px",
  "perspectiveOrigin": "632px 2px",
  "transformOrigin": "632px 2px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "0px",
  "webkitLogicalWidth": "1258px",
  "webkitPaddingAfter": "3px",
  "webkitPaddingBefore": "1px",
  "webkitPaddingEnd": "2px",
  "webkitPaddingStart": "4px",
  "width": "1258px",
}
`;

exports[`Basic ".plain-div" 1`] = `
Object {
  "blockSize": "0px",
  "cssFloat": "none",
  "display": "block",
  "height": "0px",
  "inlineSize": "1264px",
  "length": 283,
  "perspectiveOrigin": "632px 0px",
  "transformOrigin": "632px 0px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "0px",
  "webkitLogicalWidth": "1264px",
  "width": "1264px",
}
`;

exports[`Basic ".red" 1`] = `
Object {
  "backgroundColor": "rgb(255, 0, 0)",
  "blockSize": "100px",
  "cssFloat": "none",
  "display": "block",
  "height": "100px",
  "inlineSize": "100px",
  "length": 283,
  "perspectiveOrigin": "50px 50px",
  "transformOrigin": "50px 50px",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "100px",
  "webkitLogicalWidth": "100px",
  "width": "100px",
}
`;

exports[`Basic ".rgba" 1`] = `
Object {
  "blockSize": "18px",
  "borderBottomColor": "rgba(200, 200, 200, 0.5)",
  "borderLeftColor": "rgba(200, 200, 200, 0.5)",
  "borderRightColor": "rgba(200, 200, 200, 0.5)",
  "borderTopColor": "rgba(200, 200, 200, 0.5)",
  "caretColor": "rgba(200, 200, 200, 0.5)",
  "color": "rgba(200, 200, 200, 0.5)",
  "columnRuleColor": "rgba(200, 200, 200, 0.5)",
  "cssFloat": "none",
  "display": "block",
  "height": "18px",
  "inlineSize": "1264px",
  "length": 283,
  "outlineColor": "rgba(200, 200, 200, 0.5)",
  "perspectiveOrigin": "632px 9px",
  "textDecorationColor": "rgba(200, 200, 200, 0.5)",
  "transformOrigin": "632px 9px",
  "webkitBorderAfter": "0px none rgba(200, 200, 200, 0.5)",
  "webkitBorderAfterColor": "rgba(200, 200, 200, 0.5)",
  "webkitBorderBefore": "0px none rgba(200, 200, 200, 0.5)",
  "webkitBorderBeforeColor": "rgba(200, 200, 200, 0.5)",
  "webkitBorderEnd": "0px none rgba(200, 200, 200, 0.5)",
  "webkitBorderEndColor": "rgba(200, 200, 200, 0.5)",
  "webkitBorderStart": "0px none rgba(200, 200, 200, 0.5)",
  "webkitBorderStartColor": "rgba(200, 200, 200, 0.5)",
  "webkitLocale": "\\"en\\"",
  "webkitLogicalHeight": "18px",
  "webkitLogicalWidth": "1264px",
  "webkitTextEmphasisColor": "rgba(200, 200, 200, 0.5)",
  "webkitTextFillColor": "rgba(200, 200, 200, 0.5)",
  "webkitTextStrokeColor": "rgba(200, 200, 200, 0.5)",
  "width": "1264px",
}
`;
